includes/payments/actions.php

```diff
@@ -366,6 +370,12 @@ function edd_recover_payment() {
 		return;
 	}
 
+	if ( is_user_logged_in() && $payment->user_id != get_current_user_id() ) {
+		$redirect = get_permalink( edd_get_option( 'purchase_history_page' ) );
+		edd_set_error( 'edd-payment-recovery-user-mismatch', __( 'Error resuming payment.', 'easy-digital-downloads' ) );
```

Comment:
Let's add more explanation in the error, otherwise it will be had for store admins to assist customers and for them to understand why the error was shown.